internal_workflow_testsuite.go

```diff
@@ -566,6 +570,21 @@ func (env *testWorkflowEnvironmentImpl) Complete(result []byte, err error) {
 	}
 
 	close(env.doneChannel)
+
+	if env.isChildWorkflow() {
+		// this is completion of child workflow
+		childWorkflowID := env.workflowInfo.WorkflowExecution.ID
+		if childWorkflowHandle, ok := env.childWorkflows[childWorkflowID]; ok {
```

Comment:
In the case of child workflow completed after cancellation. The childWorkflowHandle would already have been removed from the map when it was request to cancel.